Question 1
* The following is a list of 10 students ages:
* ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
*  Sort the list and find the min and max age
*  Add the min age and the max age again to the list
*  Find the median age (one middle item or two middle items divided by two)
*  Find the average age (sum of all items divided by their number)
*  Find the range of the ages (max minus min)

In [3]:
ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
sorted_ages = sorted(ages)
print(sorted_ages)
print("min age : ",sorted_ages[0])
print("max age : ",sorted_ages[-1])

middle1 = sorted_ages[len(sorted_ages) // 2 - 1]
middle2 = sorted_ages[len(sorted_ages) // 2]
median = (middle1 + middle2) / 2
print("Medium : ",median)

average_age = sum(ages) / len(ages)
print("Average :",average_age)

avg = sorted_ages[-1] - sorted_ages[0]
print("Range : ",avg)

[19, 19, 20, 22, 24, 24, 24, 25, 25, 26]
min age :  19
max age :  26
Medium :  24.0
Average : 22.8
Range :  7


Question 2
* Create an empty dictionary called dog
* Add name, color, breed, legs, age to the dog dictionary
* Create a student dictionary and add first_name, last_name, gender, age, marital status,
skills, country, city and address as keys for the dictionary
* Get the length of the student dictionary
* Get the value of skills and check the data type, it should be a list
* Modify the skills values by adding one or two skills
* Get the dictionary keys as a list
* Get the dictionary values as a list

In [1]:
dog = {}

# Add name, color, breed, legs, age to the dog dictionary
dog['name'] = 'Fido'
dog['color'] = 'Brown'
dog['breed'] = 'Labrador'
dog['legs'] = 4
dog['age'] = 3

# Create a student dictionary
student = {
    'first_name': 'John',
    'last_name': 'Doe',
    'gender': 'Male',
    'age': 25,
    'marital_status': 'Single',
    'skills': ['Python', 'JavaScript'],
    'country': 'USA',
    'city': 'New York',
    'address': '123 Main St'
}

# Get the length of the student dictionary
student_length = len(student)

# Get the value of skills and check the data type
skills_value = student['skills']
skills_data_type = type(skills_value)

# Modify the skills values by adding one or two skills
skills_value.extend(['HTML', 'CSS'])

# Get the dictionary keys as a list
keys_list = list(student.keys())

# Get the dictionary values as a list
values_list = list(student.values())

# Print results
print("Dog dictionary:", dog)
print("Student dictionary:", student)
print("Length of student dictionary:", student_length)
print("Data type of skills value:", skills_data_type)
print("Modified skills value:", skills_value)
print("Keys of the student dictionary:", keys_list)
print("Values of the student dictionary:", values_list)

Dog dictionary: {'name': 'Fido', 'color': 'Brown', 'breed': 'Labrador', 'legs': 4, 'age': 3}
Student dictionary: {'first_name': 'John', 'last_name': 'Doe', 'gender': 'Male', 'age': 25, 'marital_status': 'Single', 'skills': ['Python', 'JavaScript', 'HTML', 'CSS'], 'country': 'USA', 'city': 'New York', 'address': '123 Main St'}
Length of student dictionary: 9
Data type of skills value: <class 'list'>
Modified skills value: ['Python', 'JavaScript', 'HTML', 'CSS']
Keys of the student dictionary: ['first_name', 'last_name', 'gender', 'age', 'marital_status', 'skills', 'country', 'city', 'address']
Values of the student dictionary: ['John', 'Doe', 'Male', 25, 'Single', ['Python', 'JavaScript', 'HTML', 'CSS'], 'USA', 'New York', '123 Main St']


### #3 Create a tuple containing names of your sisters and your brothers (imaginary siblings are
fine)
*  Join brothers and sisters tuples and assign it to siblings
*  How many siblings do you have?
* Modify the siblings tuple and add the name of your father and mother and assign it to
family_members

In [2]:
sisters = ('Alice', 'Emily', 'Grace')
brothers = ('Jack', 'William', 'Henry')

# Join brothers and sisters tuples and assign it to siblings
siblings = sisters + brothers

# How many siblings do you have?
num_siblings = len(siblings)

# Modify the siblings tuple and add the name of your father and mother and assign it to family_members
family_members = ('Father', 'Mother') + siblings

# Print results
print("Siblings:", siblings)
print("Number of siblings:", num_siblings)
print("Family members:", family_members)

Siblings: ('Alice', 'Emily', 'Grace', 'Jack', 'William', 'Henry')
Number of siblings: 6
Family members: ('Father', 'Mother', 'Alice', 'Emily', 'Grace', 'Jack', 'William', 'Henry')


## #4it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}
age = [22, 19, 24, 25, 26, 24, 25, 24]
* Find the length of the set it_companies
* Add 'Twitter' to it_companies
* Insert multiple IT companies at once to the set it_companies
* Remove one of the companies from the set it_companies
* What is the difference between remove and discard
* Join A and B
* Find A intersection B
* Is A subset of B
* Are A and B disjoint sets
* Join A with B and B with A
* What is the symmetric difference between A and B
* Delete the sets completely
* Convert the ages to a set and compare the length of the list and the set.

In [3]:
it_companies = {'Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon'}
A = {19, 22, 24, 20, 25, 26}
B = {19, 22, 20, 25, 26, 24, 28, 27}
age = [22, 19, 24, 25, 26, 24, 25, 24]

# Find the length of the set it_companies
length_it_companies = len(it_companies)

# Add 'Twitter' to it_companies
it_companies.add('Twitter')

# Insert multiple IT companies at once to the set it_companies
it_companies.update(['LinkedIn', 'Dell', 'Intel'])

# Remove one of the companies from the set it_companies
it_companies.remove('IBM')

# Difference between remove and discard:
# - remove() raises an error if the element is not present in the set, while discard() does not raise an error.
# - discard() silently does nothing if the element is not present in the set.
# - So, remove() is more strict than discard().

# Join A and B
joined_set = A.union(B)

# Find A intersection B
intersection_set = A.intersection(B)

# Is A subset of B
is_subset = A.issubset(B)

# Are A and B disjoint sets
disjoint_sets = A.isdisjoint(B)

# Join A with B and B with A
joined_AB = A.union(B)
joined_BA = B.union(A)

# What is the symmetric difference between A and B
symmetric_difference = A.symmetric_difference(B)

# Delete the sets completely
del it_companies
del A
del B

# Convert the ages to a set and compare the length of the list and the set
ages_set = set(age)
length_age_list = len(age)
length_age_set = len(ages_set)

## #5 The radius of a circle is 30 meters.
* Calculate the area of a circle and assign the value to a variable name of _area_of_circle_
* Calculate the circumference of a circle and assign the value to a variable name of
_circum_of_circle_
* Take radius as user input and calculate the area.


In [5]:
import math

# Given radius
radius = 30

# Calculate the area of a circle
area_of_circle = math.pi * radius ** 2

# Calculate the circumference of a circle
circum_of_circle = 2 * math.pi * radius

# Print the calculated values
print("Area of the circle:", area_of_circle)
print("Circumference of the circle:", circum_of_circle)

# Take radius as user input
user_radius = float(input("Enter the radius of the circle: "))

# Calculate the area of the circle with user input radius
area_with_user_input = math.pi * user_radius ** 2

# Print the calculated area with user input radius
print("Area of the circle with user input radius:", area_with_user_input)


Area of the circle: 2827.4333882308138
Circumference of the circle: 188.49555921538757
Area of the circle with user input radius: 50.26548245743669


## #6 “I am a teacher and I love to inspire and teach people”
* How many unique words have been used in the sentence? Use the split methods and set
to get the unique words.

In [6]:
sentence = "I am a teacher and I love to inspire and teach people"

# Split the sentence into words
words = sentence.split()

# Get unique words using set
unique_words = set(words)

# Count the number of unique words
num_unique_words = len(unique_words)

# Print the number of unique words
print("Number of unique words:", num_unique_words)

Number of unique words: 10


## #7 Use a tab escape sequence to get the following lines.
* Name Age Country City
* Asabeneh 250 Finland Helsinki


In [7]:
formatted_lines = "Name\tAge\tCountry\tCity\nAsabeneh\t250\tFinland\tHelsinki\nQuestion 8"

# Printing the formatted lines
print(formatted_lines)

# Using string formatting method to display the given statement
radius = 10
area = 3.14 * radius ** 2
formatted_statement = f"The area of a circle with radius {radius} is {area} meters square."

# Printing the formatted statement
print(formatted_statement)

Name	Age	Country	City
Asabeneh	250	Finland	Helsinki
Question 8
The area of a circle with radius 10 is 314.0 meters square.


## #8 Use the string formatting method to display the following:
* radius = 10
* area = 3.14 * radius ** 2
* “The area of a circle with radius 10 is 314 meters square.”

In [8]:
# Given radius and calculated area
radius = 10
area = 3.14 * radius ** 2

# Using string formatting method to display the statement
formatted_statement = "The area of a circle with radius {} is {} meters square.".format(radius, area)

# Printing the formatted statement
print(formatted_statement)


The area of a circle with radius 10 is 314.0 meters square.


## #9 Write a program, which reads weights (lbs.) of N students into a list and convert these weights to
kilograms in a separate list using Loop. N: No of students (Read input from user)
* Ex: L1: [150, 155, 145, 148]
* Output: [68.03, 70.3, 65.77, 67.13]

In [9]:
N = int(input("Enter the number of students: "))

# Initialize an empty list to store weights in pounds
weights_lbs = []

# Read weights of N students into the list
for i in range(N):
    weight = float(input(f"Enter weight of student {i+1} in pounds: "))
    weights_lbs.append(weight)

# Convert weights from pounds to kilograms and store them in a separate list
weights_kgs = []
for weight_lbs in weights_lbs:
    weight_kgs = weight_lbs * 0.453592  # 1 pound = 0.453592 kilograms
    weights_kgs.append(round(weight_kgs, 2))  # rounding to 2 decimal places

# Print the weights in kilograms
print("Weights in kilograms:", weights_kgs)


Weights in kilograms: [15.88, 15.42, 29.94, 24.95, 19.96, 14.97]
